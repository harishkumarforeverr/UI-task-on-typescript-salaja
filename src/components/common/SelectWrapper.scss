.conatinerCard {
  margin: 1%;
  min-width: 286px;
  min-height: 271px;
  max-height: 373px;
  height: 100%;
  background: #F2F8FB !important;
  box-shadow: 0px 6px 12px -2px rgba(0, 0, 0, 0.2) !important;
  border-radius: 6px !important;
  position: relative;
  #selectlabelCommon{
    color:black !important;
    &:hover,&:active,&:focus,&:focus-within{

      color:black !important;
    }
  }
  .ant-card-body {
    // display: grid;
    width: 286px;
    height: 326px;
    // height: 100%;
  }
  .GuiSupport {
    font-family: Helvetica;
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 24px;
    color: #3d3d3d;
    margin-left: 20px;
  }
  .deatiles {
    margin-top: 2.3rem;
  }
  .btn {
    position: absolute;
    bottom: 27px;
    align-self: end;
    left: 30px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 8px 20px;
    gap: 8px;
    width: 114px;
    height: 44px;

    background: #cc0000;
    border-radius: 2px;
    color: white;
  }
  .ColorContainer {
    width: 18px;
    height: 18px;
    left: 151px;
    top: 197px;
    border-radius: 50%;
    display: inline-block;
    // background: #cc0000;
  }
  .numberText {
    font-family: Helvetica;
    font-style: normal;
    font-weight: 400;
    font-size: 18px;
    line-height: 24px;
    margin-left: 10px;
    /* identical to box height, or 133% */

    /* Text color */

    color: #231f20;
  }
}
