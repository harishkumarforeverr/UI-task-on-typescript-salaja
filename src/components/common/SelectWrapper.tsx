import React from "react";

import { Button, Card } from "antd";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select, { SelectChangeEvent } from "@mui/material/Select";
import "./SelectWrapper.scss";
interface propsType {
  dotColor: string;
  deviceTitle: string;
  devDesc?: string;
  devSubDesc?: string;
  BtnSelectDeviceOpacity: string;
  handleChange?: (event: SelectChangeEvent) => void;
  handleTheMotor?: () => void;
}
function SelectWrapper({
  dotColor,
  deviceTitle,
  devDesc,
  BtnSelectDeviceOpacity,
  devSubDesc,
  handleChange,
  handleTheMotor,
}: propsType) {
  return (
    <Card className="conatinerCard" bordered={false} style={{ width: 300 }}>
      <div>
        <>
          <FormControl fullWidth>
            <InputLabel shrink={true} id="demo-simple-select-label">
              Select Device*
            </InputLabel>
            <Select
              notched={true}
              placeholder="Select Your Device"
              labelId="demo-simple-select-label"
              id="demo-simple-select"
              // value={age}
              label="Select Device*"
              onChange={handleChange}
            >
              <MenuItem value={10}>MCF83164EVM</MenuItem>
              <MenuItem value={20}>MCF8315EVM</MenuItem>
              <MenuItem value={30}>MCF8316EVM</MenuItem>
            </Select>
          </FormControl>
        </>
      </div>
      <div>
        <div className="deatiles">
          <span
            style={{
              background: dotColor,
            }}
            className="ColorContainer"
          ></span>
          <span className="numberText">{deviceTitle}</span>
        </div>
        <p className="GuiSupport">{devDesc}</p>
        <p
          className="GuiSupport"
          style={{
            fontSize: "12px",
            opacity: "0.8",
          }}
        >
          {devSubDesc}
        </p>
      </div>
      <Button
        style={{
          opacity: BtnSelectDeviceOpacity,
        }}
        className="lightRed btn"
        onClick={handleTheMotor}
      >
        Select Device
      </Button>
    </Card>
  );
}

export default SelectWrapper;
