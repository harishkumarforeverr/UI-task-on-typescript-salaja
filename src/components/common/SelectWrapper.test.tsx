import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import SelectWrapper from "./SelectWrapper";

const mockHistoryPush = jest.fn();
const mockGoBack = jest.fn();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/",
  }),
  useHistory: () => ({
    push: mockHistoryPush,
    goBack: mockGoBack,
  }),
  useParams: () => ({
    pathParam: "haribusine-HOAoFA",
  }),
  useNavigate: () => ({
    pathParam: "haribusine-HOAoFA",
  }),
}));

const handleChange = jest.fn((arg: any) => arg);
const handleTheMotor = jest.fn(() => null);
const BtnSelectDeviceOpacity = "0.5";
const dotColor = "#40C36C";
const deviceTitle = "MCF8316EVM";
const devDesc = "This GUI supports MCF8316EVM";
const devSubDesc = "Correct device detected : MCF8316A";
 
test("renders the compoenet and finding whether a text present in the dom or not", () => {
  const MySelectWrapper = () => {
    return (
      <Router>
        <SelectWrapper
          handleChange={handleChange}
          handleTheMotor={handleTheMotor}
          BtnSelectDeviceOpacity={BtnSelectDeviceOpacity}
          dotColor={dotColor}
          deviceTitle={deviceTitle}
          devDesc={devDesc}
          devSubDesc={devSubDesc}
        />
      </Router>
    );
  };
  render(<MySelectWrapper />);
  const linkElement = screen.getByText("This GUI supports MCF8316EVM");
  expect(linkElement).toBeInTheDocument();
});

it("creating the snapshot of the compoenent correctly", () => {
  const MySelectWrapper = () => {
    return (
      <Router>
        <SelectWrapper
          handleChange={handleChange}
          handleTheMotor={handleTheMotor}
          BtnSelectDeviceOpacity={BtnSelectDeviceOpacity}
          dotColor={dotColor}
          deviceTitle={deviceTitle}
          devDesc={devDesc}
          devSubDesc={devSubDesc}
        />
      </Router>
    );
  };
  const tree = renderer.create(<MySelectWrapper />).toJSON();
  expect(tree).toMatchSnapshot();
});
