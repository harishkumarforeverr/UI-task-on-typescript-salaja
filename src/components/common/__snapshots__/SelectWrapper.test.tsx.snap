// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating the snapshot of the compoenent correctly 1`] = `
<div
  className="ant-card conatinerCard css-dev-only-do-not-override-1n7nwfa"
  style={
    Object {
      "width": 300,
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          id="selectlabelCommon"
        >
          Select Device*
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-4gwozz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="selectlabelCommon demo-simple-select"
            className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="demo-simple-select"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Select Your Device
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="Select Your Device"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-14lo706"
            >
              <span>
                Select Device*
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        className="deatiles"
      >
        <span
          className="ColorContainer"
          style={
            Object {
              "background": "#40C36C",
            }
          }
        />
        <span
          className="numberText"
        >
          MCF8316EVM
        </span>
      </div>
      <p
        className="GuiSupport"
      >
        This GUI supports MCF8316EVM
      </p>
      <p
        className="GuiSupport"
        style={
          Object {
            "fontSize": "12px",
            "opacity": "0.8",
          }
        }
      >
        Correct device detected : MCF8316A
      </p>
    </div>
    <button
      className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-default lightRed btn"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "opacity": "0.5",
        }
      }
      type="button"
    >
      <span>
        Select Device
      </span>
    </button>
  </div>
</div>
`;
